src/api/v2/createAzureResourcesHostApi.ts

```diff
@@ -42,5 +43,15 @@ export function createAzureResourcesHostApi(
             workspaceResourceBranchDataProviderManager.addProvider(type, provider);
             return new vscode.Disposable(() => workspaceResourceBranchDataProviderManager.removeProvider(type));
         },
+
+        revealAzureResource: (id: string, options?: VSCodeRevealOptions) => {
+            // TODO: don't create new telemetry context here, instead use the context created from the initial wrapped API call
+            return callWithTelemetryAndErrorHandling('internalRevealResource', context => {
+                context.errorHandling.rethrow = true;
+                context.errorHandling.suppressDisplay = true;
+                context.errorHandling.suppressReportIssue = true;
+                return revealResource(context, id, options);
+            });
+        },
```

Comment:
More info on this todo...

We wrap methods on the API with telemetry handling and I want to be able to access the telemetry context created there here. The best solution I found was async hooks. I'm open for suggestions on what to do.

Reply:
This would effectively be reusing an action context for multiple events...can't say I'm a fan.

Reply:
IMO a shared correlation ID would be reasonable. Certainly easier to match up than timestamps.

Reply:
Ok, I don't think it's necessary now. But that's an option if we need to add correlation later. I don't see us needing to know the answer to "Which client extensions are revealing which resources?".

Reply:
That's fair. I think we'll already know the answer anyway.